### DISCUSSION TITLE
How to display Chinese characters in UTF-16

### DISCUSSION THREAD

**Discussion by twinklestar03** [Vector35/binaryninja-api#6214] [Q&A] [answered]
Binary ninja seems failed to display Chinese characters in UTF-16
I've followed [a discussion](https://github.com/Vector35/binaryninja-api/discussions/5949) related to Chinese unicode to setup codepage properly.

Here's the raw data:
```
2cf5a899c48  data_2cf5a899c48:
2cf5a899c48  c3 53 78 65 78 65 3c 50  .Sxexe<P
2cf5a899c50  70 75 38 5e 21 00 00 00  pu8^!...
```

After marked it as `wchar16`, it become something weird:
```
2cf5a899c48  wchar16 data_2cf5a899c48[0x8] = "\xcf\x83"
2cf5a899c48      "\xd5\xb8\xd5\xb8\xc0\xbc\xd5\xb0\xf8\xb8!", 0
```

If I patched the string into UTF-8 everything works.
<img width="413" alt="image" src="https://github.com/user-attachments/assets/15bf3386-6544-4cae-a49c-d7a712dd33b0">

I used Python for the conversion: 
`bytes.fromhex('c35378657865cf917075385e21000000').decode('utf16').encode('utf8')`

**Comment by negasora** [comment, marked as answer]
This was an issue with decoding of UTF16 strings, fixed in `4.3.6524-dev` tracked by https://github.com/Vector35/binaryninja-api/issues/6226
